src/chrome/internal/breakpoints/features/breakpointsEvents.ts

```diff
@@ -0,0 +1,21 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+import { NotificationChannelIdentifier } from '../../../communication/notificationsCommunicator';
+import { registerChannels } from '../../../communication/channel';
+import { BPRecipeInSource } from '../bpRecipeInSource';
+import { CDTPBPRecipe, CDTPBreakpoint } from '../../../cdtpDebuggee/cdtpPrimitives';
+import { BPRecipeIsUnbound } from '../bpRecipeStatusForRuntimeLocation';
+
+const _breakpointsEvents = {
```

Comment:
I don't really understand this stuff, would this be replaced by DI in theory in the future or is this something else?

Reply:
My current plan is to try to use the DI framework with different components that are used from different layers in the application, and we want to hide the exact implementation, and also for components that might need to be overwritten by chrome-debug, or node-debug.

These events are used to communicate between different sub-components of the breakpoint feature, which are more coupled with each other than all the components we use in the DI framework.

In summary my plan is to:
Use the DI framework for composing different components from different layers
Use this event dispatcher code to communicate inside a single "logical" component where the sub-components are more coupled.

I'm happy to talk more about this, if this approach doesn't make sense to you.

Reply:
It sounds like this stuff fills the same purpose as DI, just within one feature area instead of globally. Maybe I don't really understand how it works but it seems like it would be simpler to just use the same DI infrastructure. If you don't want all the detailed breakpoints components to be accessible globally, other DI systems I've seen have some sort of concept of a DI "container" that you can register in instead of registering them globally. Then it would be like Breakpoints owns its own private set of injectable components without introducing new terms and concepts for someone to learn.

Reply:
I don't know whether inversify has something like that

Reply:
It seems inversify has something that might work: https://github.com/inversify/InversifyJS/blob/master/wiki/hierarchical_di.md
Is it okay if I investigate this, and do this change after we merge this PR?

Reply:
Sure